### PR TITLE
Improve error messages from compliance fetcher

### PR DESCRIPTION
**Pull request by stevendanna**
Signed-off-by: Steven Danna steve@chef.io

### CODE REVIEW

**Comment by chris-rock** [comment]
I am not convinced that a fetcher should exit the application. It may throw an exception, but it should never exit the application. If used as a library (as in audit cookbook), the fetcher is not executed as cli application. From my perspective only the CLI application should decide if an exit code should be thrown or not

File: lib/inspec/ui.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# encoding: utf-8
+module Inspec
+  module UI
+    def self.exit_with_error(message)
+      $stderr.puts message
+      exit(1)
```
Comment (chris-rock) [inline lib/inspec/ui.rb]: I strongly advice against to exit application from an internal method.

File: lib/bundles/inspec-compliance/target.rb [hunk 2]
```diff
@@ -24,11 +25,23 @@ def self.resolve(target)
 
       # check if we have a compliance token
       config = Compliance::Configuration.new
-      return nil if config['token'].nil?
+      if config['token'].nil?
+        Inspec::UI.exit_with_error <<EOF
```
Comment (chris-rock) [inline lib/bundles/inspec-compliance/target.rb]: Its a great idea to add better information about an error, but that should be encapsulated in exceptions

**Review by chris-rock — CHANGES_REQUESTED** [review]

**Comment by stevendanna** [comment]
@chris-rock Yeah, I am not thrilled with exiting from here either.  One of the issues with the exception is not dumping the stack trace which obscures the error message.

**Comment by stevendanna** [comment]
Updated per code review

**Review by chris-rock — APPROVED** [review]

**Pull request merged**